cmd/push.go

```diff
@@ -87,11 +104,40 @@ Example - Push an SPDX SBOM to a registry with annotations and credentials
 				os.Exit(1)
 			}
 
-			err = obom.PushSBOM(sbom, desc, bytes, opts.reference, annotations, resolver, opts.pushSummary)
+			repo, err := getRemoteRepoTarget(opts.reference, resolver)
+			if err != nil {
+				fmt.Println("Error getting remote repository:", err)
+				os.Exit(1)
+			}
+
+			fmt.Printf("Pushing SBOM to %s@%s...\n", opts.reference, desc.Digest)
+			subject, err := obom.PushSBOM(sbom, desc, bytes, opts.reference, annotations, opts.pushSummary, repo)
 			if err != nil {
 				fmt.Println("Error pushing SBOM:", err)
 				os.Exit(1)
 			}
+			fmt.Printf("SBOM pushed to %s@%s\n", opts.reference, subject.Digest)
+
+			// attach artifacts if any
+			if len(attachArtifacts) > 0 {
+				for artifactType, paths := range attachArtifacts {
+					for _, path := range paths {
+						// load the artifact from the path
+						artifactDesc, artifactBytes, err := obom.LoadArtifactFromFile(path, artifactType)
+						if err != nil {
+							fmt.Println("Error loading artifact:", err)
+							os.Exit(1)
+						}
+						fmt.Printf("Attaching artifact %s with artifactType %s...\n", path, artifactType)
+						_, err = obom.AttachArtifact(subject, artifactDesc, artifactType, artifactBytes, repo)
+						if err != nil {
+							fmt.Println("Error attaching artifact:", err)
+							os.Exit(1)
+						}
+						fmt.Printf("Artifact attached at %s@%s\n", opts.reference, subject.Digest)
+					}
+				}
+			}
```

Comment:
Looks like we are attaching artifact one-by-one. This works, but since we call `Copy` multiple times, there might be many roundtrips of HEAD requests for checking content existence. Another way I'm thinking of is:

1. Push the subject to the memory store
2. Tag the subject in the memory store
3. For each artifact file,
    a. push it to the memory store using [`oras.PushBytes`](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0#PushBytes)
    b. pack a manifest with subject (from step 1) and the layer (from step 3.a)
4. Call [`oras.ExtenedCopy`](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0#ExtendedCopy) on the subject, which would copy the subject along with all the attached referrers.

Reply:
@Wwwsylvia I refactored to use the logic you proposed here by moving this logic into `PushSBOM`. Let me know if this is what you were thinking.